driver-core/src/main/com/mongodb/internal/connection/ConcurrentPool.java

```diff
@@ -141,7 +141,7 @@ public T get() {
     @Override
     public T get(final long timeout, final TimeUnit timeUnit) {
         if (closed) {
-            throw new IllegalStateException("The pool is closed");
+            throw poolClosedException();
```

Comment:
`DefaultConnectionPool` must report that is is closed the same way `ConcurrentPool` does. Throwing different exceptions about the same condition just because a connection pool is implemented as two classes is confusing. I had to do the same thing in https://github.com/mongodb/mongo-java-driver/pull/701 (pausable pool), and additionally had to introduce `ConcurrentPool.isPoolClosedException(Throwable)`.

Reply:
My bad, didn't notice it was called from elsewhere now.